c/tskit/tables.c

```diff
@@ -9327,35 +9329,63 @@ tsk_table_collection_check_mutation_integrity(
             ret = TSK_ERR_MUTATION_PARENT_EQUAL;
             goto out;
         }
-        /* Check if time is nonfinite */
+        /* Check if time is nonfinite and less recent than node time */
         mutation_time = mutations.time[j];
         unknown_time = tsk_is_unknown_time(mutation_time);
         if (!unknown_time) {
             if (!isfinite(mutation_time)) {
                 ret = TSK_ERR_TIME_NONFINITE;
                 goto out;
             }
+            if (mutation_time < node_time[mutations.node[j]]) {
+                ret = TSK_ERR_MUTATION_TIME_YOUNGER_THAN_NODE;
+                goto out;
+            }
+        }
+
+        /* reset checks if needed */
```

Comment:
Wasn't immediately obvious what the "if needed" meant here. Maybe "Reset counters for the next site". Also, can make it a bit more concise by
```c
if (j > 0 && mutations.site[j - 1] != mutations.site[j]) {
    ...
}
```
this is guaranteed to be safe as short circuiting booleans is part of the C standard.